test/pkg/test/framework.go

```diff
@@ -315,8 +315,42 @@ func (env *Framework) GetTargets(ctx context.Context, targetGroup *vpclattice.Ta
 	return found
 }
 
+func (env *Framework) IsVpcAssociatedWithServiceNetwork(ctx context.Context, vpcId string, serviceNetwork *vpclattice.ServiceNetworkSummary) (bool, error) {
+	Logger(ctx).Infof("IsVpcAssociatedWithServiceNetwork vpcId:%v serviceNetwork: %v \n", vpcId, serviceNetwork)
+	vpcAssociations, err := env.LatticeClient.ListServiceNetworkVpcAssociationsAsList(ctx, &vpclattice.ListServiceNetworkVpcAssociationsInput{
+		ServiceNetworkIdentifier: serviceNetwork.Id,
+		VpcIdentifier:            &vpcId,
+	})
+	if err != nil {
+		return false, err
+	}
+	if len(vpcAssociations) != 1 {
+		return false, fmt.Errorf("Expect to have one VpcServiceNetworkAssociation len(vpcAssociations): %d", len(vpcAssociations))
+	}
+	association := vpcAssociations[0]
+	if *association.Status != vpclattice.ServiceNetworkVpcAssociationStatusActive {
+		return false, fmt.Errorf("Current cluster should have one Active status association *association.Status: %s, err: %w", *association.Status, err)
+	}
+	return true, nil
+}
+
+func (env *Framework) AreAllLatticeTargetsHealthy(ctx context.Context, tg *vpclattice.TargetGroupSummary) (bool, error) {
+	Logger(ctx).Infof("Checking whether AreAllLatticeTargetsHealthy for targetGroup: %v", tg)
+	targets, err := env.LatticeClient.ListTargetsAsList(ctx, &vpclattice.ListTargetsInput{TargetGroupIdentifier: tg.Id})
+	if err != nil {
+		return false, err
+	}
+	for _, target := range targets {
+		Logger(ctx).Infof("Checking target: %v", target)
+		if *target.Status != vpclattice.TargetStatusHealthy {
+			return false, fmt.Errorf("Target is not healthy yet, target: %v", target)
```

Comment:
I think err should be nil here, since call is successful but target is not healthy yet. If you still want to return error here, then you can remove bool, since True happens only when there are no errors, which means err==nil

Reply:
changed